### PR TITLE
Validate grok pattern name on the client to avoid duplicate names.

### LINKED ISSUE #937: Mark content that was imported from content packs
Somehow mark that a stream, dashboard, ... was created by importing a content pack. That would give a great usability benefit. Maybe with a little (!) icon or something?

The backend could just store a `from_content_pack:true` flag with the created entity.

### PR DESCRIPTION
**Pull request by bernd**
Implements suggested solutions for Graylog2/graylog2-server#937.

@edmundoa Please check if that's the way to go or if I should do it different. Thanks!

### CODE REVIEW

File: javascript/src/components/grok-patterns/GrokPatterns.jsx [hunk 1]
```diff
@@ -61,6 +61,10 @@ var GrokPatterns = React.createClass({
                 </tbody>
             </table>);
     },
+    validPatternName(name) {
+        // Check if patterns already contain a pattern with the given name.
+        return this.state.patterns.map((p) => p.name).indexOf(name) === -1;
+    },
```
Comment (edmundoa) [inline javascript/src/components/grok-patterns/GrokPatterns.jsx]: I'd use a [`some` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) there, as you don't need to go through the whole array when there's a duplicated element.

File: javascript/src/components/grok-patterns/EditPatternModal.jsx [hunk 2]
```diff
@@ -9,21 +9,34 @@ var EditDashboardModal = React.createClass({
         return {
             id: this.props.id,
             name: this.props.name,
-            pattern: this.props.pattern
+            pattern: this.props.pattern,
+            _error: false,
+            _error_message: ""
```
Comment (edmundoa) [inline javascript/src/components/grok-patterns/EditPatternModal.jsx]: You can remove the `_` prefix from error and error message, in the end is the component status and it's private :)

**Comment by edmundoa** [comment]
I really like what I see, great job!

In case you want to add some validation to the pattern itself, I'd pass a single validation function to `EditPatternModal` to keep it clean. In a more complex form, passing several validation functions as properties could become really messy. Anyway, for now that is not a problem :+1:

File: javascript/src/components/grok-patterns/GrokPatterns.jsx (hunk 1 above)
Comment (bernd) [inline javascript/src/components/grok-patterns/GrokPatterns.jsx]: Nice trick, thanks!

**Comment by bernd** [comment]
CI will fail because it depends on Graylog2/graylog2-server#1034

**Comment by bernd** [comment]
@edmundoa This is complete now. Please check the latest additions. Thank you!

Needs Graylog2/graylog2-server#1034 to be merged.

**Comment by edmundoa** [comment]
Looks good to me, great job! :+1:

**Pull request merged**